Docker creates files under `target` as root, breaking future `mvn clean` runs

The first run succeeds because there is no `target` folder, but the second hits:
```
wtwhite@wtwhite-vuw-vm:~/code/jcompile$ time ./docker-build-project.sh eclipse-temurin:8u372-b07-jdk openjdk-8.0.372 commons-io commons-io-2.7.jar rel/commons-io-2.7 jars
docker image: eclipse-temurin:8u372-b07-jdk
docker container basename: openjdk-8.0.372
docker container name: openjdk-8.0.372.1098906
project name: commons-io
project jar to be generated: commons-io-2.7.jar
project tag: rel/commons-io-2.7
result root folder: jars
result folder: jars/openjdk-8.0.372

checking out tag rel/commons-io-2.7
HEAD is now at 6efbccc8 Update POM version numbers for Apache Commons IO release 2.7
using data folder /home/wtwhite/code/jcompile/dataset
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------------< commons-io:commons-io >------------------------
[INFO] Building Apache Commons IO 2.7
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ commons-io ---
[INFO] Deleting /home/wtwhite/code/jcompile/dataset/commons-io/target
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.373 s
[INFO] Finished at: 2023-10-17T13:52:39+13:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:3.1.0:clean (default-clean) on project commons-io: Failed to clean project: Failed to delete /home/wtwhite/code/jcompile/dataset/commons-io/target/antrun/build-main.xml -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
8u372-b07-jdk: Pulling from library/eclipse-temurin
--snip--
wtwhite@wtwhite-vuw-vm:~/code/jcompile$ ls -l /home/wtwhite/code/jcompile/dataset/commons-io/target/antrun/build-main.xml 
-rw-r--r-- 1 root root 326 Oct 17 13:52 /home/wtwhite/code/jcompile/dataset/commons-io/target/antrun/build-main.xml
```

--- a/docker-build-project.sh
+++ b/docker-build-project.sh
@@ -88,6 +88,7 @@ docker run \
 	--volume ${MAVEN_CACHE_HOST}:${MAVEN_CACHE_CONTAINER} \
 	--volume ${MAVEN_HOST}:${MAVEN_CONTAINER} \
 	--workdir $PROJECT2BUILD \
+	--user $(id -u):$(id -g) \
 	--name $DOCKER_CONTAINER $DOCKER_IMAGE \
 
 echo "building project"

